From 396f82798a4d19b57fc9933490dc1bd7624c9bbb Mon Sep 17 00:00:00 2001
From: Lenny Komow <lenny@lunarg.com>
Date: Mon, 25 Jun 2018 13:59:47 -0600
Subject: [PATCH] headers: Remove loader/layer dependency

The header vk_layer.h included vk_layer_dispatch_table.h, which is not
present in this repo. Instead, it had to be generated from either the
Vulkan-Loader or Vulkan-ValidationLayers repo. This change removes
that dependency.
---
 include/vulkan/vk_layer.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/include/vulkan/vk_layer.h b/include/vulkan/vk_layer.h
index e8300c3..823c88a 100644
--- a/include/vulkan/vk_layer.h
+++ b/include/vulkan/vk_layer.h
@@ -35,9 +35,6 @@
 #define VK_LAYER_EXPORT
 #endif
 
-// Definition for VkLayerDispatchTable and VkLayerInstanceDispatchTable now appear in externally generated header
-#include "vk_layer_dispatch_table.h"
-
 #define MAX_NUM_UNKNOWN_EXTS 250
 
  // Loader-Layer version negotiation API.  Versions add the following features:
@@ -50,6 +47,9 @@
 
 #define VK_CURRENT_CHAIN_VERSION 1
 
+// Typedef for use in the interfaces below
+typedef PFN_vkVoidFunction (VKAPI_PTR *PFN_GetPhysicalDeviceProcAddr)(VkInstance instance, const char* pName);
+
 // Version negotiation values
 typedef enum VkNegotiateLayerStructType {
     LAYER_NEGOTIATE_UNINTIALIZED = 0,

